pkg/libvmi/vm.go

```diff
@@ -59,8 +59,17 @@ func NewVirtualMachine(vmi *v1.VirtualMachineInstance, opts ...VMOption) *v1.Vir
 	return vm
 }
 
+func WithRunStrategy(runStrategy v1.VirtualMachineRunStrategy) VMOption {
+	return func(vm *v1.VirtualMachine) {
+		vm.Spec.RunStrategy = &runStrategy
+	}
+}
+
 func WithRunning() VMOption {
 	return func(vm *v1.VirtualMachine) {
+		// RunStrategy and running are mutually exclusive, because they can be contradictory
+		// The API server will reject VirtualMachine resources that define both
+		vm.Spec.RunStrategy = nil
```

Comment:
It is better not to have this, but if this is requiring too many changes on other tests just mention it to and mention it is an exception.

Reply:
Right, so it's just that there are too many WithRunning invocations
```bash
$ (cd tests; git grep "WithRunning" | wc -l)
29
```
and I think we want to keep them for the reason that we can't simply abandon Running, have to ensure it doesn't break.
I have a longer version of this explanation in the commit message:
https://github.com/kubevirt/kubevirt/pull/12023/commits/30a0671593813ae5f8b005227fe35938d323ebfd

WDYT about putting a comment on WithRunning that discourages it's use, along with the hidden detail of setting runStrategy to nil?

Reply:
Or, we could also have a hacky WithRunning variation outside of libvmi, something like WithDiscouragedRunning under libvmifact

Reply:
done, let me know what you think

Reply:
Please squash that last commit into the one that introduced this change.

Reply:
done